registry/tests/routerDomains.spec.ts

```diff
@@ -0,0 +1,225 @@
+import _ from 'lodash';
+import { request, expect, requestWithAuth } from './common';
+
+const example = {
+    url: '/api/v1/router_domains/',
+    correct: Object.freeze({
+        domainName: 'domainNameCorrect',
+    }),
+    updated: Object.freeze({
+        domainName: 'domainNameUpdated',
+    }),
+};
+
+describe(`Tests ${example.url}`, () => {
+    describe('Create', () => {
+        it('should not create record without a required fields', async () => {
+            await request.post(example.url)
+                .send(_.omit(example.correct, ['domainName']))
+                .expect(422, '"domainName" is required');
+        });
+
+        it('should not create record with incorrect type of fields', async () => {
+            const incorrect = {
+                domainName: 123,
+            };
+
+            await request.post(example.url)
+                .send(incorrect)
+                .expect(422, '"domainName" must be a string');
+        });
+
+        it('should successfully create record', async () => {
+            const responseCreation = await request.post(example.url)
```

Comment:
ensure cleanup in case of test failure with try/finally. Here and in all tests below

Reply:
done